src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -425,6 +427,70 @@ OverlayImpl::checkStopped ()
         stopped();
 }
 
+static
+void
+prepareValidatorKeyManifests (ManifestCache& mc,
+                              BasicConfig const& config,
+                              DatabaseCon& db,
+                              beast::Journal const& journal)
+{
+    auto const validator_keys = config.section ("validator_keys");
+    auto const validation_manifest = config.section ("validation_manifest");
+
+    if (! validator_keys.lines().empty())
+    {
+        for (auto const& line : validator_keys.lines())
+        {
+            mc.configValidatorKey (line, journal);
+        }
+    }
+    else
+    {
+        if (journal.warning)
+            journal.warning << "[validator_keys] is empty";
+    }
+
+    if (! validation_manifest.lines().empty())
+    {
+        std::string s;
+        for (auto const& line : validation_manifest.lines())
+            s += beast::rfc2616::trim(line);
+        s = beast::base64_decode(s);
+        if (auto mo = make_Manifest (std::move (s)))
+        {
+            mc.configManifest (std::move (*mo), journal);
+        }
+        else
+        {
+            throw std::runtime_error("Malformed manifest in config");
+        }
+    }
+    else
+    {
+        if (journal.warning)
+            journal.warning << "No [validation_manifest] section in config";
+    }
+
+    mc.add (db, journal);
+}
+
+void
+OverlayImpl::setupValidatorKeyManifests (BasicConfig const& config,
+                                         DatabaseCon& db)
+{
+    prepareValidatorKeyManifests (manifestCache_,
```

Comment:
Why do we need to call `prepareValidatorKeyManifests` instead of just inlining its body directly here?